src/cargo/core/resolver/encode.rs

```diff
@@ -281,14 +281,15 @@ impl<'a, 'cfg> Encodable for WorkspaceResolve<'a, 'cfg> {
 
         let root = self.ws.members().max_by_key(|member| {
             member.name()
-        }).unwrap().package_id();
+        }).map(Package::package_id);
```

Comment:
Yes, that panic is triggered when running `test` on a virtual manifest. It was currently prevented by the "no targets specified in the manifest" error.